actors/miner/tests/util.rs

```diff
@@ -2279,9 +2284,341 @@ fn fixed_hasher(offset: ChainEpoch) -> Box<dyn Fn(&[u8]) -> [u8; 32]> {
+            state.locked_funds
+        ),
+    );
+
+    // non zero funds implies that DeadlineCronActive is true
+    if state.continue_deadline_cron() {
+        acc.require(state.deadline_cron_active, "DeadlineCronActive == false when IP+PCD+LF > 0");
+    }
+}
+
+fn check_precommits<BS: Blockstore>(
+    policy: &Policy,
+    state: &State,
+    store: &BS,
+    allocated_sectors: &BTreeSet<u64>,
+    acc: &MessageAccumulator,
+) {
+    let quant = state.quant_spec_every_deadline(policy);
+
+    // invert pre-commit clean up queue into a lookup by sector number
+    let mut cleanup_epochs: BTreeMap<u64, ChainEpoch> = BTreeMap::new();
+    match BitFieldQueue::new(store, &state.pre_committed_sectors_cleanup, quant) {
+        Ok(queue) => {
+            let ret = queue.amt.for_each(|epoch, expiration_bitfield| {
+                let epoch = epoch as ChainEpoch;
+                let quantized = quant.quantize_up(epoch);
+                acc.require(
+                    quantized == epoch,
+                    &format!("pre-commit expiration {epoch} is not quantized"),
+                );
+
+                expiration_bitfield.iter().for_each(|sector_number| {
+                    cleanup_epochs.insert(sector_number, epoch);
+                });
+                Ok(())
+            });
+            acc.require_no_error(ret, "error iterating pre-commit clean-up queue");
+        }
+        Err(e) => {
+            acc.add(&format!("error loading pre-commit clean-up queue: {e}"));
+        }
+    };
+
+    let mut precommit_total = BigInt::zero();
+
+    let precommited_sectors =
+        Map::<_, SectorPreCommitOnChainInfo>::load(&state.pre_committed_sectors, store);
+
+    match precommited_sectors {
+        Ok(precommited_sectors) => {
+            let ret = precommited_sectors.for_each(|key, precommit| {
+                let sector_number = match parse_uint_key(&key) {
+                    Ok(sector_number) => sector_number,
+                    Err(e) => {
+                        acc.add(&format!("error parsing pre-commit key as uint: {e}"));
+                        return Ok(());
+                    }
+                };
+
+                acc.require(
+                    allocated_sectors.contains(&sector_number),
+                    &format!("pre-commited sector number has not been allocated {sector_number}"),
+                );
+
+                acc.require(
+                    cleanup_epochs.contains_key(&sector_number),
+                    &format!("no clean-up epoch for pre-commit at {}", precommit.pre_commit_epoch),
+                );
+                precommit_total += &precommit.pre_commit_deposit;
+                Ok(())
+            });
+            acc.require_no_error(ret, "error iterating pre-commited sectors");
+        }
+        Err(e) => {
+            acc.add(&format!("error loading precommited_sectors: {e}"));
+        }
+    };
+
```

Comment:
Nice to have assuming its easy is to assert equality on the size of precommitted_sectors and cleanup_epochs.  This way we check the invariant that all cleanup sector numbers correspond to a precommit.

Reply:
@ZenGround0 Hm, I added a piece of code to check that but it fails several tests with non-zero clean-up epochs and zero precommited sectors len
```
            let mut precommited_sectors_count = 0;
            let ret = precommited_sectors.for_each(|key, precommit| {
                precommited_sectors_count += 1;
                ...
            });
            acc.require(cleanup_epochs.len() == precommited_sectors_count, &format!("cleanup epochs: {}, precommited sectors len: {precommited_sectors_count}", cleanup_epochs.len()));
```
e.g.
```
--- compacting_a_partition_with_both_live_and_dead_sectors_removes_dead_sectors_retains_live_sectors stdout ----
thread 'compacting_a_partition_with_both_live_and_dead_sectors_removes_dead_sectors_retains_live_sectors' panicked at 'cleanup epochs: 4, precommited sectors len: 0', actors/miner/tests/util.rs:2596:5
```
Am I missing something?